2008-05-10  Ulrich Drepper  <drepper@redhat.com>

	* sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_rdlock.S: Access
	__pshared correctly.
	* sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedrdlock.S:
	Likewise.
	* sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedwrlock.S:
	Likewise.
	* sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_unlock.S:
	Likewise.
	* sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_wrlock.S:
	Likewise.
	Reported by Clemens Kolbitsch <clemens.kol@gmx.at>.

--- libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_rdlock.S	14 Aug 2007 02:24:49 -0000	1.15
+++ libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_rdlock.S	11 May 2008 05:40:33 -0000	1.16
@@ -122,7 +122,7 @@ __pthread_rwlock_rdlock:
 #else
 	leal	MUTEX(%ebx), %edx
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_lock_wait
 	jmp	2b
 
@@ -138,7 +138,7 @@ __pthread_rwlock_rdlock:
 #else
 	leal	MUTEX(%ebx), %eax
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_unlock_wake
 	jmp	7b
 
@@ -158,7 +158,7 @@ __pthread_rwlock_rdlock:
 #else
 	leal	MUTEX(%ebx), %eax
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_unlock_wake
 	jmp	11b
 
@@ -168,7 +168,7 @@ __pthread_rwlock_rdlock:
 #else
 	leal	MUTEX(%ebx), %edx
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_lock_wait
 	jmp	13b
 	.size	__pthread_rwlock_rdlock,.-__pthread_rwlock_rdlock
--- libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedrdlock.S	14 Aug 2007 02:25:44 -0000	1.16
+++ libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedrdlock.S	11 May 2008 05:40:33 -0000	1.17
@@ -162,7 +162,7 @@ pthread_rwlock_timedrdlock:
 #else
 	leal	MUTEX(%ebp), %edx
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_lock_wait
 	jmp	2b
 
@@ -177,7 +177,7 @@ pthread_rwlock_timedrdlock:
 #else
 	leal	MUTEX(%ebp), %eax
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_unlock_wake
 	jmp	7b
 
@@ -197,7 +197,7 @@ pthread_rwlock_timedrdlock:
 #else
 	leal	MUTEX(%ebp), %eax
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_unlock_wake
 	jmp	11b
 
@@ -207,7 +207,7 @@ pthread_rwlock_timedrdlock:
 #else
 	leal	MUTEX(%ebp), %edx
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_lock_wait
 	jmp	13b
 
--- libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedwrlock.S	14 Aug 2007 02:24:58 -0000	1.17
+++ libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_timedwrlock.S	11 May 2008 05:40:33 -0000	1.18
@@ -160,7 +160,7 @@ pthread_rwlock_timedwrlock:
 #else
 	leal	MUTEX(%ebp), %edx
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_lock_wait
 	jmp	2b
 
@@ -175,7 +175,7 @@ pthread_rwlock_timedwrlock:
 #else
 	leal	MUTEX(%ebp), %eax
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_unlock_wake
 	jmp	7b
 
@@ -190,7 +190,7 @@ pthread_rwlock_timedwrlock:
 #else
 	leal	MUTEX(%ebp), %eax
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_unlock_wake
 	jmp	11b
 
@@ -200,7 +200,7 @@ pthread_rwlock_timedwrlock:
 #else
 	leal	MUTEX(%ebp), %edx
 #endif
-	movl	PSHARED(%ebp), %ecx
+	movzbl	PSHARED(%ebp), %ecx
 	call	__lll_lock_wait
 	jmp	13b
 
--- libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_unlock.S	14 Aug 2007 02:25:06 -0000	1.13
+++ libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_unlock.S	11 May 2008 05:40:33 -0000	1.14
@@ -110,7 +110,7 @@ __pthread_rwlock_unlock:
 #else
 	leal	MUTEX(%edi), %edx
 #endif
-	movl	PSHARED(%edi), %ecx
+	movzbl	PSHARED(%edi), %ecx
 	call	__lll_lock_wait
 	jmp	2b
 
@@ -120,7 +120,7 @@ __pthread_rwlock_unlock:
 #else
 	leal	MUTEX(%edi), %eax
 #endif
-	movl	PSHARED(%edi), %ecx
+	movzbl	PSHARED(%edi), %ecx
 	call	__lll_unlock_wake
 	jmp	4b
 
@@ -130,7 +130,7 @@ __pthread_rwlock_unlock:
 #else
 	leal	MUTEX(%edi), %eax
 #endif
-	movl	PSHARED(%edi), %ecx
+	movzbl	PSHARED(%edi), %ecx
 	call	__lll_unlock_wake
 	jmp	8b
 
--- libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_wrlock.S	14 Aug 2007 02:25:27 -0000	1.15
+++ libc/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_rwlock_wrlock.S	11 May 2008 05:40:33 -0000	1.16
@@ -120,7 +120,7 @@ __pthread_rwlock_wrlock:
 #else
 	leal	MUTEX(%ebx), %edx
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_lock_wait
 	jmp	2b
 
@@ -135,7 +135,7 @@ __pthread_rwlock_wrlock:
 #else
 	leal	MUTEX(%ebx), %eax
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_unlock_wake
 	jmp	7b
 
@@ -149,7 +149,7 @@ __pthread_rwlock_wrlock:
 #else
 	leal	MUTEX(%ebx), %eax
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_unlock_wake
 	jmp	11b
 
@@ -159,7 +159,7 @@ __pthread_rwlock_wrlock:
 #else
 	leal	MUTEX(%ebx), %edx
 #endif
-	movl	PSHARED(%ebx), %ecx
+	movzbl	PSHARED(%ebx), %ecx
 	call	__lll_lock_wait
 	jmp	13b
 	.size	__pthread_rwlock_wrlock,.-__pthread_rwlock_wrlock
